### PR TITLE
Adds blog post on scale testing pulp_ansible

### PR DESCRIPTION
**Pull request by bmbouter**
This is viewable in rendered form here: https://bmbouter.github.io/pulpproject.org/2020/03/04/pulp-3-scale-testing/

### CODE REVIEW

File: _posts/2020-03-05-pulp-3-scale-testing.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+other operations on any of the set of Repositories in the batch will not execute in parallel with
+other Pulp workloads.
+
+Each batch's set of Repositories are mutually exclusive with all other batches so these tasks are
+fully parallelizable with each other. This means the more workers a Pulp system has the workload
+should run faster and faster, assuming adequate hardware resources.
+
+**NOTE: Anytime we talk about Repository, we technically are talking about an [`AnsibleRepository`](
+https://pulp-ansible.readthedocs.io/en/latest/restapi.html#operation/
+repositories_ansible_ansible_read) since Pulp repositories are typed.
```
Comment (daviddavis) [inline _posts/2020-03-05-pulp-3-scale-testing.md:69]: Unclosed `**` here.

File: _posts/2020-03-05-pulp-3-scale-testing.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+* `run2` - The amount of time in seconds the second run took to process all tasks.
+* `run3` - The amount of time in seconds the third run took to process all tasks.
+* `avg time` - The average amount of time from the three runs.
+* `avg repos / second` - The average amount of RepositoryVersion's created per second.
+
+
+## Database and CPU Constrained
+
+The system's load falls mostly onto the DB and is CPU bound. All 16 workers of Pulp were engaged
+(as expected) and the postgresql processes auto-scales in terms of it's processes, but it only
```
Comment (daviddavis) [inline _posts/2020-03-05-pulp-3-scale-testing.md]: s/it's/its/

File: _posts/2020-03-05-pulp-3-scale-testing.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+* `batch size` - The number of repositories handled per task.
+* `num_tasks` - The number of tasks to be handled by the Pulp workers. Note the batch sizes are
+                selected to generate a number of tasks that is divisible by 16, the number of
+                workers this system has.
+* `run1` - The amount of time in seconds the first run took to process all tasks.
+* `run2` - The amount of time in seconds the second run took to process all tasks.
+* `run3` - The amount of time in seconds the third run took to process all tasks.
+* `avg time` - The average amount of time from the three runs.
+* `avg repos / second` - The average amount of RepositoryVersion's created per second.
```
Comment (daviddavis) [inline _posts/2020-03-05-pulp-3-scale-testing.md]: RepositoryVersions

File: _posts/2020-03-05-pulp-3-scale-testing.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+This system used 16 Pulp workers and so the work was divided over 16 parallel processes. If the DB
+has enough resources to keep up and Pulp had more cores it could start more workers and they could
+also run in parallel.
+
+This means that those with more resources can scale up as much as their PostgreSQL cluster will.
+Also if you want to start a few workers Pulp will just work on the same workload over more time.
+
+An average RepositoryVersions per second is roughly 31.4 based on all the runs in the table. With
+16 workers, that is 31.4 / 16 =  1.96 ~ 2 repositories per second. To estimate the capability of
```
Comment (daviddavis) [inline _posts/2020-03-05-pulp-3-scale-testing.md]: This sentence is confusing. Probably want `created` after `RepositoryVersions`.

File: _posts/2020-03-05-pulp-3-scale-testing.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+With each run more RepositoryVersions are created. Before the table was generated it was already
+173,311. With each test run that number increased and for the 10,000 repo promotion test it was
+increasing 5% each time (roughly). This causes a non-negligible slowdown which you can see in the
+10,000 repository row, with each test taking an additional 8 seconds.
+
+
+## Notes & Resources
+
+To run this scale test I needed to develop some tools. I added these to pulp_ansible and documented
+them [here](https://pulp-ansible. readthedocs.io/en/latest/testing.html#scale-testing-tools).
```
Comment (daviddavis) [inline _posts/2020-03-05-pulp-3-scale-testing.md]: Bad link I think.

File: _posts/2020-03-05-pulp-3-scale-testing.md (hunk 1 above)
Comment (bmbouter) [reply _posts/2020-03-05-pulp-3-scale-testing.md:69]: ty, fixed in next push.

File: _posts/2020-03-05-pulp-3-scale-testing.md (hunk 2 above)
Comment (bmbouter) [reply _posts/2020-03-05-pulp-3-scale-testing.md]: ty, fixed in next push.

File: _posts/2020-03-05-pulp-3-scale-testing.md (hunk 3 above)
Comment (bmbouter) [reply _posts/2020-03-05-pulp-3-scale-testing.md]: ty, fixed in next push.

**Review by daviddavis — APPROVED** [review]

File: _posts/2020-03-05-pulp-3-scale-testing.md (hunk 4 above)
Comment (bmbouter) [reply _posts/2020-03-05-pulp-3-scale-testing.md]: ty, rewritten in next push.

File: _posts/2020-03-05-pulp-3-scale-testing.md (hunk 5 above)
Comment (bmbouter) [reply _posts/2020-03-05-pulp-3-scale-testing.md]: fixed in next push, ty

**Pull request merged**